embedded-graphics/src/draw_target/mod.rs

```diff
@@ -439,6 +447,161 @@ pub trait DrawTarget {
     /// [`size`]: #method.size
     /// [`fill_solid`]: #method.fill_solid
     fn clear(&mut self, color: Self::Color) -> Result<(), Self::Error> {
-        self.fill_solid(&Rectangle::new(Point::zero(), self.size()), color)
+        self.fill_solid(&self.bounding_box(), color)
+    }
+}
+
+/// Extension trait for `DrawTarget`s.
+pub trait DrawTargetExt: DrawTarget + Sized {
+    /// Creates a translated draw target based on this draw target.
+    ///
+    /// All drawing operations are translated by `offset` pixels, before being passed to the base
+    /// draw target.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use embedded_graphics::{
+    ///     prelude::*,
+    ///     mock_display::MockDisplay,
+    ///     pixelcolor::BinaryColor,
+    ///     fonts::{Text, Font6x8},
+    ///     style::TextStyle,
+    /// };
+    ///
+    /// let mut display = MockDisplay::new();
+    /// let mut translated_display = display.translated(Point::new(10, 5));
+    ///
+    /// // Draws text at position (10, 5) in the display coordinate system
+    /// Text::new("Text", Point::zero())
+    ///     .into_styled(TextStyle::new(Font6x8, BinaryColor::On))
+    ///     .draw(&mut translated_display)?;
+    /// #
+    /// # let mut expected = MockDisplay::new();
+    /// #
+    /// # Text::new("Text", Point::new(10, 5))
+    /// #     .into_styled(TextStyle::new(Font6x8, BinaryColor::On))
+    /// #     .draw(&mut expected)?;
+    /// #
+    /// # assert_eq!(display, expected);
+    /// #
+    /// # Ok::<(), core::convert::Infallible>(())
+    /// ```
+    fn translated(&mut self, offset: Point) -> Translated<'_, Self>;
+
+    /// Creates a cropped draw target based on this draw target.
+    ///
+    /// A cropped draw target is a draw target for a rectangular subregion of the base draw target.
+    /// Its coordinate system is shifted so that the origin coincides with `area.top_left` in the
+    /// base draw target's coordinate system.
+    ///
+    /// The bounding box of the returned target will always be contained inside the bounding box
+    /// of the base target. If the requested `area` is overlapping the base target's bounding box
+    /// the intersection of the base target's bounding box and `area` will be used.
```

Comment:
```suggestion
    /// The bounding box of the returned target will always be contained inside the bounding box
    /// of the base target. If any of the requested `area` lies outside the base target's bounding 
    /// box the intersection of the base target's bounding box and `area` will be used.
```

Reply:
Thanks for the suggestion, I was struggling to get this paragraph right.